programs/fileio.c

```diff
@@ -1033,6 +1034,8 @@ FIO_compressGzFrame(const cRess_t* ress,  /* buffers & handlers are used, but no
             EXM_THROW(79, "zstd: %s: deflateEnd error %d \n", srcFileName, ret);
     }   }
     *readsize = inFileSize;
+    AIO_WritePool_releaseIoJob(writeJob);
+    AIO_WritePool_sparseWriteEnd(ress->writeCtx);
```

Comment:
We don't want to be doing sparse writes here, since we should basically never have long runs of zeros in a compressed file.

Do we always create the compression-side `writeCtx` with `sparseFileSupport == 0`?

Reply:
We create the `writeCtx` with the default `sparseFileSupport` for compression, which afaik is the global default (1 for all archs apart from MacOS).
This can be fixed by updating `sparseFileSupport` to 0 in the case of compression. Do you think it will make any measurable impact?

Reply:
I'd guess yes. But it would be worth benchmarking.

There is also no reason to do it, so setting it to 0 in the case of compression makes sense to me. And compression doesn't currently use sparse writes, so there is no change in behavior.

Reply:
Fixed, now setting to 0 in `zstcli.c` for compression.